### ISSUE TITLE
[A11ySev2] [Programmatic Access-Dashboards with Grafana – Azure Insights Key Vaults]: Ensure elements with ARIA roles have all required ARIA attributes.

### ISSUE THREAD

**Issue by LinglingTong** [grafana/grafana#118768]
## Steps to reproduce

Repro Steps: 
1. Open 'Azure Monitor dashboards with Grafana' in Azure Portal
2. "Monitor | Dashboards with Grafana | Gallery" screen appears with already Dashboards with Grafana tab selected.  
3. Select already existing data as 'Azure Insights Key Vaults'  under All Dashboards tab.  
4. Navigate all control under Newley opened page like Tool bar , Controls under tool bar (dataSource , Subscription , Resource group,...etc) and overview section. 
5. Run Accessibility Insight Tool and Verify the issue. 

**Actual Result:**
The elements with the role="heading" do not have the required aria-level attribute. As a result, screen readers cannot determine the heading level, which makes it harder for users to understand the page's structure or navigate using heading modes.  
**Observation**:  
The span elements with the role="heading" do not have the aria-level attribute, which is a required ARIA attribute for elements with the heading role. This results in screen readers being unable to identify the heading level, negatively impacting navigation for users with disabilities. 
**Issue 1:** Same issue repro for the row title controls after adding the row from add menu button in new dashboard.  
**Navigation**: Open https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGrafanaReactView  > Select New button button > Invoke New dashboard control> Select “ Add visualization” button > Select testdata list item> Select Back to dashboard in the tool bar > Select Add row from add menu button in the tool bar  > Run ai tool and verify the issue.   **Expected Result: ** 
All elements with the ARIA role heading must include the aria-level attribute to indicate their heading level within the page hierarchy. For example, the "Overview," "Failures," and "Operations" elements should have an aria-level attribute with appropriate values (e.g., aria-level="1" for top-level headings or aria-level="2" for subheadings).  

 
**Issue**: Ensure elements with ARIA roles have all required ARIA attributes (aria-required-attr - https://accessibilityinsights.io/info-examples/web/aria-required-attr)  
   
**Target application:** Monitor - Microsoft Azure - https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGrafanaReactView  
   
**

**Relevant WCAG Criteria:** [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) (Level A)

## Environment

- **Grafana version:** 12.2.0
- **Data source type & version:**
- **User OS & Browser:** Version 25H2 (OS Build 26200.7093) / Chrome Version 142.0.7444.177 (Official Build) (64-bit)
- **Others:**

**Comment by LinglingTong** [comment]
<img width="2843" height="942" alt="Image" src="https://github.com/user-attachments/assets/cf9b498a-5e2a-4dce-a58e-3c22fbd8072b" />
<img width="2822" height="950" alt="Image" src="https://github.com/user-attachments/assets/ee042ca5-f915-497e-88c3-a252269be3a9" />

**Comment by ashharrison90** [comment]
this is fixed with the new dynamic dashboards that are currently rolling out - we no longer set a `role="heading"` on row titles
